src/components/Post/GovernanceSideBar/Referenda/VoteReferendum.tsx

```diff
@@ -81,14 +83,19 @@ const VoteReferendum = ({ className, referendumId, onAccountChange, lastVote, se
 	const [vote, setVote] = useState< EVoteDecisionType>(EVoteDecisionType.AYE);
 
 	useEffect(() => {
-		if(!window) return;
-		const Wallet = localStorage.getItem('loginWallet') ;
-		if(Wallet){
-			setLoginWallet(Wallet as  Wallet);
-			setWallet(Wallet as Wallet);
+		if (userDetails.loginWallet) {
+			setLoginWallet(userDetails.loginWallet);
+			setWallet(userDetails.loginWallet);
+		} else {
+			if(!window) return;
+			const Wallet = localStorage.getItem('loginWallet') ;
```

Comment:
not a required change but var names ideally shouldn't start with a capital letter.